nilearn/plotting/displays.py

```diff
@@ -829,13 +829,19 @@ def add_markers(self, marker_coords, marker_color='r', marker_size=30,
             coord = display_ax.coord
             marker_coords_2d, third_d = _coords_3d_to_2d(
                 marker_coords, direction, return_direction=True)
-            # Heuristic that plots only markers that are 2mm away from the
-            # current slice.
-            # XXX: should we keep this heuristic?
-            mask = np.abs(third_d - coord) <= 2.
             xdata, ydata = marker_coords_2d.T
-            display_ax.ax.scatter(xdata[mask], ydata[mask],
-                                  s=marker_size,
+            # Check if coord has integer represents a cut in direction
+            # to follow the heuristic. If no foreground image is given
+            # coordinate is empty or None. This case is valid for plotting
+            # markers on glass brain without any foreground image.
+            if isinstance(coord, numbers.Number):
+                # Heuristic that plots only markers that are 2mm away
+                # from the current slice.
+                # XXX: should we keep this heuristic?
+                mask = np.abs(third_d - coord) <= 2.
```

Comment:
Not fully sure I undertsand the logic. In particular is mask contains several slices ?

Reply:
>In particular is mask contains several slices ?

mask is an array contains bool of shape (1,). Bool depends on the heuristic.

As a side note: This logic was already there. Just that this PR checks ```coord is not None``` and doesn't throws error when no image is provided (in this case ```coord is None```). The same reason I explained as a comment.

Reply:
Yes. Well, it seems to be fine anyway.
LGTM overall.